source/common/network/connection_impl.cc

```diff
@@ -539,6 +540,13 @@ void ConnectionImpl::doConnect() {
       ENVOY_CONN_LOG(debug, "immediate connection error: {}", *this, errno);
     }
   }
+
+  try {
+    local_address_ = Address::addressFromFd(fd_);
+  } catch (const EnvoyException& e) {
+    // Eat the exception and leave local_address unchanged.  This will fail
+    // when connecting to a UDS.
```

Comment:
If this is the case, can we avoid the syscall, etc. here? Is there a way for us to know via the remote address that it is UDS and just skip this?

Reply:
Maybe, but we'd have to keep this code in sync with all the cases that can throw an error in what it is calling.  I think it's simpler and more maintainable to leave it this way.

Reply:
There are two main types of errors. Errors that can happen due to user input, and errors that should basically never happen. I would put this into the latter category. This is a fast path and if we don't need the error handling we shouldn't have it (vs. just letting the process crash for example). It also just makes the code more complicated. So the question is do we need this error handling in any normal use case?

Reply:
I see your point.

I could change it to:

  if (remote_address_->type() == Address::Type::Ip) {
    local_address_ = Address::addressFromFd(fd_);
  }

This saves the syscall for UDS.  But it's probably slower (very slightly) overall for Type::Ip.  But it better conforms to Envoy's error handling strategy.

Reply:
Yup +1 (then I think you can remove the new exception throwing code you added to the address utilities). Thank you.